### PR TITLE
Improved windows_explorer.xml file

### PR DESCRIPTION
**Pull request by Tobias-B-Besemer**

### CODE REVIEW

File: cleaners/windows_explorer.xml [hunk 1]
```diff
@@ -50,14 +53,52 @@
     <action command="winreg" path="HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\Map Network Drive MRU"/>
     <!--  printer ports -->
     <action command="winreg" path="HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\PrnPortsMRU"/>
+
+
+    <!--  more MRUs -->
+
+    <!-- Completely deleted by CCleaner !!!
+    <action command="winreg" path="HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\ComDlg32"/>
+    -->
```
Comment (az0) [inline cleaners/windows_explorer.xml]: Please elaborate on this commented out action and your comment

**Comment by az0** [comment]
I marked this for milestone 2.4 (instead of 2.2) to avoid delaying the 2.2 release any more. This could use careful review and testing.

File: cleaners/windows_explorer.xml (hunk 1 above)
Comment (Tobias-B-Besemer) [reply cleaners/windows_explorer.xml]: Was just a note as a kind of documentation... This line was for my opinion to hard, so I don't included it... Maybe a later editor of the cleaner will include this line... So he have "some docs" from me to it...

File: cleaners/windows_explorer.xml [hunk 2]
```diff
@@ -18,79 +18,127 @@
...
+    <action command="winreg" path="HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\ComDlg32\CIDSizeMRU"/>
+    <action command="winreg" path="HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\ComDlg32\FirstFolder"/>
+    <action command="winreg" path="HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\ComDlg32\LastVisitedPidlMRULegacy"/>
+    <!-- End! -->
+    <action command="winreg" path="HKCU\Software\Microsoft\Windows\CurrentVersion\Explorer\ComputerDescriptions"/>
+    <action command="winreg" path="HKLM\Software\Microsoft\Direct3D\MostRecentApplication" name="Name"/>
+    <action command="winreg" path="HKCU\Software\Microsoft\Direct3D\MostRecentApplication" name="Name"/>
+    <action command="winreg" path="HKLM\Software\Microsoft\DirectDraw\MostRecentApplication"/>
+    <action command="winreg" path="HKCU\Software\Microsoft\DirectInput\MostRecentApplication" name="Id"/>
+    <action command="winreg" path="HKCU\Software\Microsoft\DirectInput\MostRecentApplication" name="Name"/>
```
Comment (az0) [inline cleaners/windows_explorer.xml]: What do Direct3D, DirectDraw, and DirectInput have to do with Windows Explorer?

File: cleaners/windows_explorer.xml [hunk 3]
```diff
@@ -18,79 +18,127 @@
...
-    <action command="process" wait="false" cmd="%WINDIR%\explorer.exe %WINDIR%\explorer.exe"/>
+    <action command="process" wait="false" cmd="%WinDir%\explorer.exe %WINDIR%\explorer.exe"/>
+    <action command="delete" search="walk.files" path="%LocalAppData%\Microsoft\Windows\Explorer\ThumbCacheToDelete\"/>
+    <action command="delete" search="file" path="%LocalAppData%\IconCache.db"/>
+    <action command="delete" search="file" path="%LocalAppData%\IconCache.db.backup"/>
+  </option>
+  <option id="mountpoints2">
+    <label>Mount Points</label>
+    <description>Delete the data</description>
+    <warning>Use the option only if you know what you do!</warning>
```
Comment (az0) [inline cleaners/windows_explorer.xml]: It isn't immediately obvious to me what this does, but it sounds scary, so maybe it shouldn't be here.

File: cleaners/windows_explorer.xml (hunk 2 above)
Comment (Tobias-B-Besemer) [reply cleaners/windows_explorer.xml]: I used winsys.ini as a reference...

2019-07-17 02_53_48-winsys ini - Editor

File: cleaners/windows_explorer.xml (hunk 3 above)
Comment (Tobias-B-Besemer) [reply cleaners/windows_explorer.xml]: https://www.techwalla.com/articles/how-to-delete-mountpoints2-with-regedit

**Comment by az0** [comment]
Thank you for your work putting this together. However, please let's not use CCleaner's `winsys.ini` as a reference for any pull requests here, so I need to close this pull request. You may resubmit a new PR with that part removed or based on another reference.

**Comment by Tobias-B-Besemer** [comment]
I used winsys.ini as a reverence, but verified got all on my long time used testing machine (~10 years) with "reverse engineering" e.g. the registry keys. So all implemented cleaning was verified and tested on my machine.
OK, I understand that using other cleaners as inspiration to look on special things in the system is not so good... but as I have now the knowledge from analysing, I can use this knowledge, right?
So I would do a split into two cleaners, one "Explorer" and one "DirectX"... Would this work for you?

**Pull request closed without merging**